### ISSUE TITLE
why we should need this function:def __patch_instance_norm_state_dict()

### ISSUE THREAD

**Issue by TJJTJJTJJ** [junyanz/pytorch-CycleGAN-and-pix2pix#441]
>    def __patch_instance_norm_state_dict(self, state_dict, module, keys, i=0):
        key = keys[i]
        if i + 1 == len(keys):  # at the end, pointing to a parameter/buffer
            if module.__class__.__name__.startswith('InstanceNorm') and \
                    (key == 'running_mean' or key == 'running_var'):
                if getattr(module, key) is None:
                    state_dict.pop('.'.join(keys))
            if module.__class__.__name__.startswith('InstanceNorm') and \
               (key == 'num_batches_tracked'):
                state_dict.pop('.'.join(keys))
        else:
            self.__patch_instance_norm_state_dict(state_dict, getattr(module, key), keys, i + 1)

What's the point of the function? In my test, InstanceNorm.running_mean and running_var are always None, we maybe don't need to the function or there's something I have ignored?

Thank you, sir.

**Comment by junyanz** [comment]
You can ignore it. It is used to load models trained with previous PyTorch versions. InstanceNorm can have running_mean and running_var when `track_running_stats=False`.

**Comment by TJJTJJTJJ** [comment]
Thank you sir.
Q1: In the paper, you write c7s1-32, but in your code , ngf=64, maybe somewhere I have confused?
> parser.add_argument('--ndf', type=int, default=64, help='# of discrim filters in first conv layer')
model = [nn.ReflectionPad2d(3),
                 nn.Conv2d(input_nc, ngf, kernel_size=7, padding=0, bias=use_bias)
                 norm_layer(ngf),
                 nn.ReLU(inplace=True)]

**Comment by TJJTJJTJJ** [comment]
Q2: In the d64 and d128, you don't use ReflectionPad, right? As I know, in the neural_style, it use ReflectionPad in the downsampling? or maybe the difference don't influence the results?

**Comment by junyanz** [comment]
We didn't use it in the original paper. It might help. Feel free to try it.

**Comment by TJJTJJTJJ** [comment]
sir how about Q1
Thank you sir.
Q1: In the paper, you write c7s1-32, but in your code , ngf=64, maybe somewhere I have confused?

    parser.add_argument('--ndf', type=int, default=64, help='# of discrim filters in first conv layer')
    model = [nn.ReflectionPad2d(3),
    nn.Conv2d(input_nc, ngf, kernel_size=7, padding=0, bias=use_bias)
    norm_layer(ngf),
    nn.ReLU(inplace=True)]

**Comment by junyanz** [comment]
It should be 64. We will update the appendix in a future version.
